packages/venia-concept/src/components/FilterModal/filterModal.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from 'react';
+import FilterFooter from './FilterFooter';
+import PropTypes from 'prop-types';
+import { List } from '@magento/peregrine';
+import { FiltersCurrent } from './FiltersCurrent';
+import classify from 'src/classify';
+import CloseIcon from 'react-feather/dist/icons/x';
+import Icon from 'src/components/Icon';
+import FilterBlock from './filterBlock';
+import defaultClasses from './filterModal.css';
+
+class FilterModal extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            root: PropTypes.string,
+            modalWrapper: PropTypes.string,
+            header: PropTypes.string,
+            headerTitle: PropTypes.string,
+            filterOptionsContainer: PropTypes.string
+        }),
+        filters: PropTypes.arrayOf(
+            PropTypes.shape({
+                request_var: PropTypes.string,
+                items: PropTypes.array
+            })
+        ),
+        filterAdd: PropTypes.func,
+        filterRemove: PropTypes.func,
+        closeModalHandler: PropTypes.func
+    };
+
+    getFilterParams = location => {
+        const params = new URLSearchParams(location.search);
+        let titles,
+            values = [];
+
+        let urlFilterParams = {};
+
+        for (var key of params.keys()) {
+            const cleanKey = key.replace(/\[\]\[.*\]/gm, '');
+
+            if (urlFilterParams[cleanKey]) continue;
+
+            titles = params.getAll(`${cleanKey}[][title]`);
+            values = params.getAll(`${cleanKey}[][value]`);
+
+            urlFilterParams[cleanKey] = titles.map((title, index) => ({
+                title: title,
+                value: values[index]
+            }));
+        }
+
+        return urlFilterParams;
+    };
+
+    componentDidMount = () => {
+        const filterParams = this.getFilterParams(this.props.history.location);
```

Comment:
I think we need to store the filters in a single location such as redux and then use that to create the chosen filters within the modal. I think the store could initialize with the values from the URL but I don't think the component should do this every time.

Reply:
Great suggestion. Resolved in latest update. Let me know if everything works as expected.